code/entities/chair/Armrest.cs

```diff
@@ -0,0 +1,98 @@
+﻿using Sandbox;
+
+namespace Cinema;
+
+public partial class Armrest : BaseNetworkable
+{
+    public enum States : int
+    {
+        Lowered,
+        Raising,
+        Raised,
+        Lowering
+    }
+
+    public enum Sides : int
+    {
+        Left,
+        Right
+    }
+
+    [Net]
+    public CinemaChair Chair { get; set; }
+    [Net]
+    public ModelEntity Cupholded { get; set; }
+    [Net]
+    public States State { get; set; } = States.Lowered;
+    [Net]
+    public Sides Side { get; set; }
+
+    public string CupholderBoneName => Side switch
+    {
+        Sides.Left  => "cupholder_l",
+        _           => "cupholder_r"
+    };
+
+    public string ArmrestToggleParameter => Side switch
+    {
+        Sides.Left => "toggle_left_armrest",
+        _ => "toggle_right_armrest"
+    };
+
+    public void HandleAnimTag(string tag)
+    {
+        if (tag.Contains("launch"))
+        {
+            Launch();
+            State = States.Raised;
+        }
+        else if (tag.Contains("lowered"))
+        {
+            State = States.Lowered;
+        }
+    }
+
+    public void Toggle()
+    {
+        var originalValue = Chair.GetAnimParameterBool(ArmrestToggleParameter);
+        Chair.SetAnimParameter(ArmrestToggleParameter, !originalValue);
+        // If toggle_seat was true but now is false, we are now lowering the seat.
+        State = originalValue
+            ? States.Lowering
+            : States.Raising;
+    }
+
+    public void CupholdEntity(ModelEntity entity)
```

Comment:
Would be more clear if this was `SetCupholdEntity` or maybe `SetCupholderEntity` or perhaps `Hold`/`HoldEntity` (was a bit confused what this was by it's name, I read it as "Cuphold an entity" but cuphold is not a verb)

Maybe even `PutEntityInCupholder` if you want to be verbose